src/nepiada/utils/graphs.py

```diff
@@ -7,38 +7,53 @@
 
 class Graph:
     def __init__(self, config, agents, screen=None, cell_size=0):
-        print("Graphs have been initialized")
-
         self.dim = config.size
         self.agents = agents
         self.dynamic_obs = config.dynamic_obs
-        self.full_communication = config.full_communication
+        self.dynamic_comms = config.dynamic_comms
+        self.dynamic_comms_radius = config.dynamic_comms_radius
+        self.dynamic_comms_enforce_minimum = config.dynamic_comms_enforce_minimum
         self.observation_radius = config.obs_radius
         self.num_agents = config.num_good_agents + config.num_adversarial_agents
         self.obs_arrows = []
         self.screen = screen
         self.clock = pygame.time.Clock()
         self.cell_size = cell_size
-        self.screen_width = config.screen_width 
+        self.screen_width = config.screen_width
         self.screen_height = config.screen_height
+
         ## An adjacency list for which agent can communicate with each other
-        if self.full_communication:
+        if not self.dynamic_comms:
             all_agents = [agent for agent in self.agents]
             self.comm = {agent: all_agents for agent in self.agents}
+
+            print(
```

Comment:
Implementing logging for debug messages will also help us control whether we print these messages or not depending on the type of run

Reply:
This would be a future PR tbh, these prints are just sanity prints to showcase all the modules working. The seperate render modes for debug and testing mode that Arash is working on will be able to show more "debugging" information, and we can have a future logging module if we want to have more information.